### PR TITLE
NOTIF-190 Add single build script which calls every other build script

### PR DESCRIPTION
**Pull request by bennetelli**

### CODE REVIEW

**Comment by bennetelli** [comment]
> It's missing the `.rhcicd/` since it will be called from the root
> 
> ```
> [theute@localhost notifications-backend]$ bash .rhcicd/build_deploy.sh 
> 
> /bin/bash build_deploy_backend.sh
> + /bin/bash build_deploy_backend.sh
> /bin/bash: build_deploy_backend.sh: No such file or directory
> ```
> 
> Also, the DockerFile used here doesn't seem correct:
> https://github.com/RedHatInsights/notifications-backend/blob/master/.rhcicd/build_deploy_aggregator.sh

Should be fixed now.

**Comment by theute** [comment]
It doesn't seem right, if it's using the same DockerFile (without additional parameters), then it will produce the same image for both pods

**Comment by pilhuhn** [comment]
I think it depends on the directory this is called from. 
`docker --config="$DOCKER_CONF" build -t "${IMAGE}:${IMAGE_TAG}" . -f ...`  will use the artefacts that are in the current working dir

**Comment by theute** [comment]
In this case docker/Dockerfile.jvm is the same file. `docker` is a directory at the base:
https://github.com/RedHatInsights/notifications-backend/blob/master/docker/Dockerfile-build.jvm

**Comment by bennetelli** [comment]
> In this case docker/Dockerfile.jvm is the same file. `docker` is a directory at the base:
> https://github.com/RedHatInsights/notifications-backend/blob/master/docker/Dockerfile-build.jvm

Ok, got it. Thanks for the hint. Fixed it.

**Comment by bennetelli** [comment]
@pilhuhn @theute would be great if you could take a look at it again so I can test it tomorrow morning :)

**Comment by theute** [comment]
I tried locally for the aggregator:
```
Removing intermediate container c2cd92bb9a90
 ---> d9257363e827
Step 6/15 : FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
 ---> dc080723f596
Step 7/15 : ARG JAVA_PACKAGE=java-11-openjdk-headless
 ---> Using cache
 ---> 1fc291833ef1
Step 8/15 : ARG RUN_JAVA_VERSION=1.3.8
 ---> Using cache
 ---> 571111ca6f90
Step 9/15 : ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en'
 ---> Using cache
 ---> d383a0adc2b8
Step 10/15 : RUN microdnf install openssl curl ca-certificates ${JAVA_PACKAGE}     && microdnf update     && microdnf clean all     && mkdir /deployments     && chown 1001 /deployments     && chmod "g+rwX" /deployments     && chown 1001:root /deployments     && curl https://repo1.maven.org/maven2/io/fabric8/run-java-sh/${RUN_JAVA_VERSION}/run-java-sh-${RUN_JAVA_VERSION}-sh.sh -o /deployments/run-java.sh     && chown 1001 /deployments/run-java.sh     && chmod 540 /deployments/run-java.sh     && echo "securerandom.source=file:/dev/urandom" >> /etc/alternatives/jre/lib/security/java.security
 ---> Using cache
 ---> 65f92be5d6ef
Step 11/15 : ENV JAVA_OPTIONS="-Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager"
 ---> Using cache
 ---> d0ca4407d82f
Step 12/15 : COPY --from=build --chown=1001 /home/jboss/backend/target/quarkus-app/ /deployments/
COPY failed: stat home/jboss/backend/target/quarkus-app/: file does not exist
```

https://github.com/RedHatInsights/notifications-backend/pull/442/files#diff-46f4f799ad50f937e778b498ae19b8ea615cecbc0b34302461c756bd9b78e498R37 is not copying from the right folder

File: .rhcicd/build_deploy_backend.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ mkdir -p "$DOCKER_CONF"
 
 docker --config="$DOCKER_CONF" login -u="$QUAY_USER" -p="$QUAY_TOKEN" quay.io
 docker --config="$DOCKER_CONF" login -u="$RH_REGISTRY_USER" -p="$RH_REGISTRY_TOKEN" registry.redhat.io
```
Comment (pilhuhn) [inline .rhcicd/build_deploy_backend.sh:22]: I think this is not needed. `IMAGE` above starts with `quay.io`

**Comment by bennetelli** [comment]
> I tried locally for the aggregator:
> 
> ```
> Removing intermediate container c2cd92bb9a90
>  ---> d9257363e827
> Step 6/15 : FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
>  ---> dc080723f596
> Step 7/15 : ARG JAVA_PACKAGE=java-11-openjdk-headless
>  ---> Using cache
>  ---> 1fc291833ef1
> Step 8/15 : ARG RUN_JAVA_VERSION=1.3.8
>  ---> Using cache
>  ---> 571111ca6f90
> Step 9/15 : ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en'
>  ---> Using cache
>  ---> d383a0adc2b8
> Step 10/15 : RUN microdnf install openssl curl ca-certificates ${JAVA_PACKAGE}     && microdnf update     && microdnf clean all     && mkdir /deployments     && chown 1001 /deployments     && chmod "g+rwX" /deployments     && chown 1001:root /deployments     && curl https://repo1.maven.org/maven2/io/fabric8/run-java-sh/${RUN_JAVA_VERSION}/run-java-sh-${RUN_JAVA_VERSION}-sh.sh -o /deployments/run-java.sh     && chown 1001 /deployments/run-java.sh     && chmod 540 /deployments/run-java.sh     && echo "securerandom.source=file:/dev/urandom" >> /etc/alternatives/jre/lib/security/java.security
>  ---> Using cache
>  ---> 65f92be5d6ef
> Step 11/15 : ENV JAVA_OPTIONS="-Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager"
>  ---> Using cache
>  ---> d0ca4407d82f
> Step 12/15 : COPY --from=build --chown=1001 /home/jboss/backend/target/quarkus-app/ /deployments/
> COPY failed: stat home/jboss/backend/target/quarkus-app/: file does not exist
> ```
> 
> https://github.com/RedHatInsights/notifications-backend/pull/442/files#diff-46f4f799ad50f937e778b498ae19b8ea615cecbc0b34302461c756bd9b78e498R37 is not copying from the right folder

that's strange.. I tested it locally for backend as well as for aggregator. In the root directory: docker --config=".docker" build -t "quay.io/cloudservices/notifications-aggregator:1338" . -f docker/Dockerfile-build-backend.jvm

and it worked

**Comment by theute** [comment]
Because you didn't "mvn clean" so it copied the `backend` quarkus app into the `aggregator` container

File: docker/Dockerfile-build-aggregator.jvm [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    && chown 1001:root /deployments \
+    && curl https://repo1.maven.org/maven2/io/fabric8/run-java-sh/${RUN_JAVA_VERSION}/run-java-sh-${RUN_JAVA_VERSION}-sh.sh -o /deployments/run-java.sh \
+    && chown 1001 /deployments/run-java.sh \
+    && chmod 540 /deployments/run-java.sh \
+    && echo "securerandom.source=file:/dev/urandom" >> /etc/alternatives/jre/lib/security/java.security
+
+# Configure the JAVA_OPTIONS, you can add -XshowSettings:vm to also display the heap size.
+ENV JAVA_OPTIONS="-Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager"
+
+COPY --from=build --chown=1001 /home/jboss/backend/target/quarkus-app/ /deployments/
```
Comment (theute) [inline docker/Dockerfile-build-aggregator.jvm:37]: Should be:
`COPY --from=build --chown=1001 /home/jboss/aggregator/target/quarkus-app/ /deployments/
`

**Comment by theute** [comment]
We still have to push to Quay so the quay credentials and login are still necessary

**Comment by bennetelli** [comment]
Which line can be removed? 

This?
docker --config="$DOCKER_CONF" login -u="$QUAY_USER" -p="$QUAY_TOKEN" quay.io

or this?
docker --config="$DOCKER_CONF" login -u="$RH_REGISTRY_USER" -p="$RH_REGISTRY_TOKEN" registry.redhat.io

**Comment by bennetelli** [comment]
Decided to take the quay login and remove the registry. At least that's what I understand when reading the Slack messages.

**Comment by theute** [comment]
Looks good from here, but didn't test because of current status:

**Review by josejulio — COMMENTED** [review]
I don't really know much about this part of the code but the containers build without visible errors.

**Comment by josejulio** [comment]
> Looks good from here, but didn't test because of current status:

Current status is awesome!

**Comment by bennetelli** [comment]
> > Looks good from here, but didn't test because of current status:
> 
> Current status is awesome!

Indeed :)

**Comment by bennetelli** [comment]
If everything is fine, someone should approve it :-)

**Comment by theute** [comment]
I just finished testing, seems ok now

**Pull request merged**